staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation_test.go

```diff
@@ -658,3 +663,857 @@ func TestCelCompilation(t *testing.T) {
 		})
 	}
 }
+
+// take a single rule type in (string/number/map/etc.) and return appropriate values for
+// Type, Format, and XIntOrString
+func parseRuleType(ruleType string) (string, string, bool) {
+	if ruleType == "duration" || ruleType == "date" || ruleType == "date-time" {
+		return "string", ruleType, false
+	}
+	if ruleType == intOrString {
+		return "", "", true
+	}
+	return ruleType, "", false
+}
+
+func genArrayWithRule(arrayType, rule string) func(maxItems *int64) *schema.Structural {
+	passedType, passedFormat, xIntString := parseRuleType(arrayType)
+	return func(maxItems *int64) *schema.Structural {
+		return &schema.Structural{
+			Generic: schema.Generic{
+				Type: "array",
+			},
+			Items: &schema.Structural{
+				Generic: schema.Generic{
+					Type: passedType,
+				},
+				ValueValidation: &schema.ValueValidation{
+					MaxItems: maxItems,
```

Comment:
This is misplaced, the MaxItems ofr the array should be in the ValueValidation of the array's schema, not the Item's schema.

Reply:
Right - fixed